test/integration/consul-container/upgrade/healthcheck_test.go

```diff
@@ -72,6 +70,8 @@ func TestTargetServersWithLatestGAClients(t *testing.T) {
 
 // Test health check GRPC call using Mixed (majority latest) Servers and Latest GA Clients
 func TestMixedServersMajorityLatestGAClient(t *testing.T) {
+	t.Parallel()
```

Comment:
I removed Parallel run at some point because it added to the flakiness. If we are confident it's fixed now we can put it back everywhere otherwise let's keep it with no parallel run for now.